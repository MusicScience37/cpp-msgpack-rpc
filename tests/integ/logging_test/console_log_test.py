"""Test of console logs.
"""

import pathlib
import subprocess
import typing

import approvaltests
from .create_scrubber import create_scrubber


def _verify_command_result(
    writer_path: pathlib.Path,
    args: typing.List[str],
    test_temp_dir_path: pathlib.Path,
) -> None:
    result = subprocess.run(
        args=[str(writer_path)] + args,
        cwd=str(test_temp_dir_path),
        stdout=subprocess.PIPE,
        stderr=subprocess.PIPE,
        check=False,
        encoding="utf-8",
    )

    approvaltests.verify(
        f"""exit code: {result.returncode}
stdout:
{result.stdout}
stderr:
{result.stderr}
""",
        options=approvaltests.Options().with_scrubber(create_scrubber()),
    )


def test_info_log(
    writer_path: pathlib.Path,
    test_temp_dir_path: pathlib.Path,
):
    """Test with info log level."""

    _verify_command_result(
        writer_path=writer_path,
        args=[],
        test_temp_dir_path=test_temp_dir_path,
    )


def test_trace_log(
    writer_path: pathlib.Path,
    test_temp_dir_path: pathlib.Path,
):
    """Test with trace log level."""

    _verify_command_result(
        writer_path=writer_path,
        args=["-v"],
        test_temp_dir_path=test_temp_dir_path,
    )


def test_quiet(
    writer_path: pathlib.Path,
    test_temp_dir_path: pathlib.Path,
):
    """Test of null log sink."""

    _verify_command_result(
        writer_path=writer_path,
        args=["-q"],
        test_temp_dir_path=test_temp_dir_path,
    )
