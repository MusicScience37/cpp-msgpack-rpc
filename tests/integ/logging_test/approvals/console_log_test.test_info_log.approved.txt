exit code: 0
stdout:
[<time-stamp>] [info] Information. (write_log.cpp:58, <function-name>)
[<time-stamp>] [warning] Warning. (write_log.cpp:59, <function-name>)
[<time-stamp>] [error] Error. (write_log.cpp:60, <function-name>)
[<time-stamp>] [critical] Critical. (write_log.cpp:61, <function-name>)
[<time-stamp>] [info] Log with an argument: 123 (write_log.cpp:63, <function-name>)

stderr:

